shared/wallets/sep7-confirm/index.tsx

```diff
@@ -0,0 +1,356 @@
+  recipient: string
+}
+const PaymentInfo = (props: PaymentInfoProps) => (
+  <Kb.Box2 direction="vertical" fullWidth={true}>
+    <Kb.Divider />
+    <Kb.Box2 direction="vertical" fullWidth={true} style={styles.memoContainer}>
+      <Kb.Text type="BodyTinySemibold" style={styles.headingText}>
+        Amount
+      </Kb.Text>
+      {!!props.amount && (
+        <>
+          <Kb.Text type="HeaderBigExtrabold" style={styles.purpleText}>
+            {props.amount} XLM
+          </Kb.Text>
+          <Kb.Box2 direction="vertical" fullWidth={true} gap="xtiny" gapStart={true} gapEnd={false}>
+            <Kb.Text type="BodyTinySemibold" style={styles.headingText}>
+              (Approximately {props.displayAmountFiat})
+            </Kb.Text>
+            <Kb.Text type="BodyTinySemibold" style={styles.headingText}>
+              Your primary account has {props.availableToSendNative} available to send.
+            </Kb.Text>
+          </Kb.Box2>
+        </>
+      )}
+    </Kb.Box2>
+    {!props.amount && <AssetInput />}
+    {!!props.memo && <InfoRow headerText="Memo" bodyText={props.memo} />}
+    {!!props.message && <InfoRow headerText="Message" bodyText={props.message} />}
+    {!!props.recipient && (
+      <InfoRow headerText="Recipient" bodyText={props.recipient} showStellarIcon={true} />
+    )}
+  </Kb.Box2>
+)
+
+type TxInfoProps = {
+  operations: Array<string>
+  fee: string
+  memo: string | null
+  source: string
+}
+const TxInfo = (props: TxInfoProps) => (
+  <Kb.Box2 direction="vertical" fullWidth={true}>
+    {props.operations.length > 0 &&
+      props.operations.map((op, idx) => (
+        <InfoRow key={idx + 1} headerText={`Operation ${idx + 1}`} bodyText={op} />
+      ))}
+    {!!props.fee && <InfoRow headerText="Fee" bodyText={props.fee + ' stroops'} />}
+    {!!props.memo && <InfoRow headerText="Memo" bodyText={props.memo} />}
+    {!!props.source && <InfoRow headerText="Source account" bodyText={props.source} showStellarIcon={true} />}
+  </Kb.Box2>
+)
+
+const SEP7Confirm = (props: Props) =>
+  props.loading ? (
+    <Loading onBack={props.onBack} />
+  ) : props.error ? (
+    <Error error={props.error} onBack={props.onBack} />
+  ) : (
+    <Kb.MaybePopup onClose={props.onBack}>
+      <Kb.Box2 direction="vertical" fullHeight={!Styles.isMobile} fullWidth={true} style={styles.container}>
+        <Header
+          isPayment={props.operation === 'pay'}
+          originDomain={props.originDomain}
+          onBack={props.onBack}
+        />
+        {!!props.callbackURL && <CallbackURLBanner callbackURL={props.callbackURL} />}
+        <Kb.ScrollView style={styles.scrollView} alwaysBounceVertical={false}>
+          {props.operation === 'pay' ? (
+            <PaymentInfo
+              amount={props.amount}
+              availableToSendNative={props.availableToSendNative}
+              displayAmountFiat={props.displayAmountFiat}
+              memo={props.memoType === 'MEMO_TEXT' && props.memo}
```

Comment:
Curious this passes tsx as `memo` is typed as `string | null` and this could be a boolean.